### ISSUE TITLE
Uncaught TypeError: Cannot read property 'getAdapter' of undefined

### ISSUE THREAD

**Issue by tardyp** [mulesoft/api-workbench#102]
1. ...
install api workbench
2. ...
open file from this repo:
https://github.com/econchick/ramlfications/blob/wip-general-code-cleanup/tests/data/v020examples/resources.raml

**Atom Version**: 1.4.1
**System**: Mac OS X 10.11.2
**Thrown From**: [api-workbench](https://github.com/mulesoft/api-workbench) package, v0.8.13

### Stack Trace

Uncaught TypeError: Cannot read property 'getAdapter' of undefined

```
At /Users/ptardy/myconfigs/configs/atom/packages/api-workbench/main.js:9

TypeError: Cannot read property 'getAdapter' of undefined
    at Object.l [as validate] (/Users/ptardy/myconfigs/configs/atom/packages/api-workbench/main.js:9:12186)
    at t.e.validate (/Users/ptardy/myconfigs/configs/atom/packages/api-workbench/main.js:3:17115)
    at e.validate (/Users/ptardy/myconfigs/configs/atom/packages/api-workbench/main.js:9:22527)
    at Object.l [as validate] (/Users/ptardy/myconfigs/configs/atom/packages/api-workbench/main.js:9:12143)
    at t.e.validate (/Users/ptardy/myconfigs/configs/atom/packages/api-workbench/main.js:3:17115)
    at /Users/ptardy/myconfigs/configs/atom/packages/api-workbench/main.js:9:12072
    at Array.forEach (native)
    at p (/Users/ptardy/myconfigs/configs/atom/packages/api-workbench/main.js:9:12043)
    at Object.l [as validate] (/Users/ptardy/myconfigs/configs/atom/packages/api-workbench/main.js:9:12234)
    at t.e.validate (/Users/ptardy/myconfigs/configs/atom/packages/api-workbench/main.js:3:17115)
    at /Users/ptardy/myconfigs/configs/atom/packages/api-workbench/main.js:9:12072
    at Array.forEach (native)
    at p (/Users/ptardy/myconfigs/configs/atom/packages/api-workbench/main.js:9:12043)
    at Object.l [as validate] (/Users/ptardy/myconfigs/configs/atom/packages/api-workbench/main.js:9:12234)
    at t.e.validate (/Users/ptardy/myconfigs/configs/atom/packages/api-workbench/main.js:3:17115)
    at /Users/ptardy/myconfigs/configs/atom/packages/api-workbench/main.js:9:12072
    at Array.forEach (native)
    at p (/Users/ptardy/myconfigs/configs/atom/packages/api-workbench/main.js:9:12043)
    at Object.l [as validate] (/Users/ptardy/myconfigs/configs/atom/packages/api-workbench/main.js:9:12234)
    at t.e.validate (/Users/ptardy/myconfigs/configs/atom/packages/api-workbench/main.js:3:17115)
    at p (/Users/ptardy/myconfigs/configs/atom/packages/api-workbench/main.js:43:4565)
    at u (/Users/ptardy/myconfigs/configs/atom/packages/api-workbench/main.js:43:4133)
    at y (/Users/ptardy/myconfigs/configs/atom/packages/api-workbench/main.js:43:4859)
    at Workspace.module.exports.Workspace.observeTextEditors (/Applications/Atom.app/Contents/Resources/app.asar/src/workspace.js:288:9)
    at Object.n [as initEditorObservers] (/Users/ptardy/myconfigs/configs/atom/packages/api-workbench/main.js:43:3394)
    at Object.b [as consumeLinter] (/Users/ptardy/myconfigs/configs/atom/packages/api-workbench/main.js:1:1257)
    at Provider.module.exports.Provider.provide (/Applications/Atom.app/Contents/Resources/app.asar/node_modules/service-hub/lib/provider.js:30:52)
    at /Applications/Atom.app/Contents/Resources/app.asar/node_modules/service-hub/lib/service-hub.js:55:26
    at doNTCallback0 (node.js:419:9)
    at process._tickCallback (node.js:348:13)
```

### Commands

```
  4x -2:54.1.0 core:backspace (atom-text-editor.editor.mini.is-focused)
     -2:42.9.0 core:confirm (atom-text-editor.editor.mini.is-focused)
  6x -2:32.8.0 core:backspace (atom-text-editor.editor.mini.is-focused)
```

### Config

```json
{
  "core": {
    "disabledPackages": [
      "symbols-view",
      "autocomplete-plus"
    ],
    "ignoredNames": [
      ".git",
      ".repo",
      "workdir",
      "libs",
      "node_modules",
      "build",
      "_trial_temp*",
      "*.pyc"
    ],
    "projectHome": "~/dev/bb",
    "themes": [
      "atom-dark-ui",
      "monokai-soda"
    ]
  }
}
```

### Installed Packages

```coffee
# User
Sublime-Style-Column-Selection, v1.5.1
angularjs, v0.3.4
api-workbench, v0.8.13
autocomplete-python, v1.5.1
environment, v1.1.1
highlight-line, v0.11.0
highlight-selected, v0.11.1
language-jade, v0.6.3
language-restructuredtext, v0.12.0
language-swift, v0.4.0
linter, v1.11.3
linter-coffeelint, v1.1.2
linter-cpplint, v1.0.1
linter-jshint, v1.2.2
linter-less, v2.3.3
linter-pylama, v0.2.6
linter-swiftc, v2.0.0
monokai-soda, v0.5.0
project-manager, v2.7.5
remote-edit, v1.8.24
zentabs, v0.8.6

# Dev
No dev packages
```

**Comment by sichvoge** [comment]
Hi @tardyp,

thanks for reporting your issue. I was able to reproduce your problem and tried to eliminate the cause step by step. I found the following with my first evaluation:

1) the parser seems to have problems with your schema in `trait` -> `filterable` -> `body`. Same for:

* post:/thingys
* put:/thingys/thingy-gizmos

2) it seems the parser also does not like that your example in the same trait body is defined in one line. The spec actually mandates that you use a pipe `|` and define your example on the next line.

![image](https://cloud.githubusercontent.com/assets/498114/12586651/00fe35cc-c449-11e5-99f6-20184506b7aa.png)

bringing it to the next line with a pipe symbol `|` fixes that. 

```yaml
example: |
  { "name" : "example body for trait" }
```

That is the same for:

* post:/thingys
* put:/thingys/thingy-gizmos

3) `formParameters` in `post:/gizmos` does not belong at this position. The following should be correct:

```yaml
/gizmos:
  displayName: several-gizmos
  securedBy: [ oauth_2_0 ]
  post:
    description: Post several gizmos
    body:
      formParameters:
        ids:
          displayName: Example Gizmo IDs
          type: string
          description: A comma-separated list of IDs
          required: true
          example: "gizmo1,gizmo2,gizmo3"
```

4) `uriParameters` in `get:/thingys/thingy-gizmos/{id}` is not correctly indented. It needs to be on the same level as your `get` method.

So, from my point of you we would need to investigate only the first point as it is valid JSON. I put in the same into http://jsonschemalint.com/draft4/# and it validates ok. We will come back soon, but if you do the correction in the meanwhile - that would be great ;)

**Comment by tardyp** [comment]
Hi thanks for the detailed answer!
cc @econchick who is the original author of that raml example.

I think indeed we need to fix our example to be valid raml, but also we shouldn't have api workbench throwing exception in atom :)

In order to help on the effort of implementing a raml1.0 parser, I was looking for an extensive testsuite, that implementations could use as known valid/invalid raml.
I think this would help a lot making implementations having a coherent interpretation of the spec.
Do you know of ongoing effort on that part we could base on?

**Comment by sichvoge** [comment]
Its funny that you mention that test suite. Be patient is all what I can say ;) 

With regards to API Workbench. I agree that the one with the schema should not throw an exception. We need to investigate why that is even happening in this case.

**Comment by tardyp** [comment]
Le mer. 27 janv. 2016 à 11:54, Christian Vogel <notifications@github.com> a
écrit :

> Its funny that you mention that test suite. Be patient is all what I can
> say ;)
>
can't wait.. As I am starting the 1.0 dev for ramlfications, we already
started building our own testsuite.


> With regards to API Workbench. I agree that the one with the schema should
> not throw an exception. We need to investigate why that is even happening
> in this case.
>
good to know!

>

**Comment by ddenisenko** [comment]
Fix for the exception published in 0.8.16.
